ReportCardParser/src/reportcardparser/ReportCardParserType1.java

```diff
@@ -5,32 +5,105 @@
  */
 package reportcardparser;
 
+import java.io.*;
+import java.util.*;
 /**
  * Class to handle parsing of a Type 1 report card
  * 
  * @author chris
  */
-public class ReportCardParserType1 implements IReportCardParser {
-    
-    private String filePath; // Path to the report card file
-    
+public class ReportCardParserType1 extends AbstractReportCardParser {
+
+
     /**
      * Constructor
      * 
      * @param filePath Path to the report card file
      */
     public ReportCardParserType1(String filePath) {
-        this.filePath = filePath;
+        super(filePath);
     }
-    
+
     /**
      * Parse the configured report card file
      * 
+     * @param br BufferedReader to read the file
      * @return ReportCard object containing the data from the configured report card file
      */
     @Override
-    public ReportCard parse() {
-        // REPLACE ME WITH A REAL IMPLEMENTATION!
+    public ReportCard parse(BufferedReader br) {
+        try
+        {
+            //Declare variables
+            String name = "", dob; //Student's name and date of birth
+            
+            String currentLine = br.readLine(); //Name line
+            StringTokenizer st = new StringTokenizer(currentLine);
+            st.nextToken(); //Skips "Name:" token
+            while (st.hasMoreTokens()) //Adds remaining tokens as student's name
+            {
+                name += st.nextToken();
+                name += " ";
+            }
+            name = name.trim();
+            
+            currentLine = br.readLine(); //Date of birth line
+            st = new StringTokenizer(currentLine);
+            st.nextToken(); //Skips "DOB:" token
+            dob = st.nextToken(); //Sets DOB
+            
+            currentLine = br.readLine(); //Blank line
+            
+            //More variable for mark parsing
+            int total = 0; //Total, for average
+            CourseMark[] marks = new CourseMark[8]; //Array of course marks
+            String[] courseName = new String[8]; //Array of course names
+            int[] mark = new int[8]; //Array of marks
+            String currentToken; //The current token
+            
+            for (int i = 0; i < 8; i++)
+            {
+                courseName[i] = "";
+                currentLine = br.readLine(); //Reads line
+                st = new StringTokenizer(currentLine);
+                while(true)
+                {
+                    currentToken = st.nextToken(); //Reads token
+                    try
+                    {
+                        mark[i] = Integer.parseInt(currentToken); //If the token is a number, set the mark and break
+                        break;
+                    }
+                    catch (NumberFormatException nfe) //If not, move to here and add the token to the course name
```

Comment:
Nice strategy for determining if the token is an integer or part of the course name.